beacon-chain/sync/initial-sync/service.go

```diff
@@ -80,7 +80,6 @@ func (s *Service) Start() {
 		// Wait for state to be initialized.
 		stateChannel := make(chan *feed.Event, 1)
 		stateSub := s.stateNotifier.StateFeed().Subscribe(stateChannel)
-		defer stateSub.Unsubscribe()
 		genesisSet := false
 		for !genesisSet {
```

Comment:
```suggestion
		for genesisSet := false; !genesisSet; {
```

This is really bothering me for some reason...